### ISSUE TITLE
unnecessary_wraps suggests to unwrap unit instead of omitting return value

### ISSUE THREAD

**Issue by msakuta** [rust-lang/rust-clippy#6640]
I tried this code:

```rust
fn always_ok() -> Result<(), String> {
    Ok(())
}
```

I expected that clippy suggests omitting return value at all:

```rust
fn always_ok() {
}
```

Instead, it suggests returning unit:

```
warning: this function's return value is unnecessarily wrapped by `Result`
  --> src/main.rs:22:1
   |
22 | / fn always_ok() -> Result<(), String> {
23 | |     Ok(())
24 | | }
   | |_^
   |
   = note: `#[warn(clippy::unnecessary_wraps)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_wraps
help: remove `Result` from the return type...
   |
22 | fn always_ok() -> () {
   |                   ^^
help: ...and change the returning expressions
   |
23 |     ()
   |
```

and it's stupid because if you follow the instructions it shows another lint warning:

```
warning: unneeded unit return type
  --> src/main.rs:22:15
   |
22 | fn always_ok() -> () {
   |               ^^^^^^ help: remove the `-> ()`
   |
   = note: `#[warn(clippy::unused_unit)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unused_unit

warning: unneeded unit expression
  --> src/main.rs:23:5
   |
23 |     ()
   |     ^^ help: remove the final `()`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unused_unit

warning: 2 warnings emitted
```

It should suggest omitting return value in the first place.

Tested on [Rust playgound](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=4a8f81c008662f29f5fbbba95222cd91)

### Meta

- `cargo clippy -V`: e.g. clippy 0.1.51 (2021-01-19 c5a96fb)
- `rustc 1.51.0-nightly`

**Comment by pag4k** [comment]
I'll try to do this issue.
